haystack/utils/app_loading.py

```diff
@@ -14,6 +15,14 @@ def haystack_get_app_modules():
     """Return the Python module for each installed app"""
     return [i.module for i in apps.get_app_configs()]
 
+def haystack_get_search_indexes():
+    """Return all modules to collect search indexes in"""
+    app_modules = [(app_mod, '%s.search_indexes' % app_mod.__name__) for app_mod in haystack_get_app_modules()]
+    custom_search_indexes = getattr(settings, 'HAYSTACK_SEARCH_INDEX_MODULES', [])
+    app_modules.extend([(None, search_index) for search_index in custom_search_indexes])
```

Comment:
Hi @SteveByerly thanks for the review, I will look into it this week.

Setting the first item to `None` was necessary to correctly re-raise the ImportError in `collect_indexes`. But to be perfectly honest, now that I think about it, I am completely unsure how the workflow is and should be. 😅 

I'll take some time this week to review the PR and update according to your review. 😄